OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -247,12 +247,10 @@ protected virtual bool CanFire(Actor self, Target target)
 			return true;
 		}
 
-		// Note: facing is only used by the legacy positioning code
-		// The world coordinate model uses Actor.Orientation
-		public virtual Barrel CheckFire(Actor self, IFacing facing, Target target)
+		public virtual bool CheckFire(Actor self, Target target)
```

Comment:
`CheckFire` has always been a weird name. Now that we're returning a bool, can we rename this to `TryFire` to better match standard conventions?